### Name: Emily Lucy

# Homework 1

#  *__Introduction__*

*Homework 1: 10 question assignment with various types of problems programmers could run into while working with Python involving variables, types of data, and operations.* 




---


# __Question 1: How much is this precipitation as an average intensity expressed in metric units in “millimeter per day”?__

###    A series of storms is expected to hit the western coast of the US this week: “[…] This combination of Pacific storms is likely to wring out heavy rain from the coastal ranges of far northwestern California to western Oregon and western Washington. Some of these areas could pick up […] to 10 inches of rain over the next seven days […]” [https://www.wunderground.com/article/forecast/regional/news/2020-01-23-pacific-storm-parade-atmospheric-rivers-washington-oregon Jan 24th, 2020]. 


In [1]:
10/7*.0254*1000

36.285714285714285

*The average intensity over 10 days is 36.285714285714285 millimeters per day.*

# __Question 2: Do the calculation with Python using two variables: one is assigned the precipitation value in inches, the other is assigned to the result from the conversion calculation.__

In [2]:
 # Conversion Calculation with Variables
x=10
y=x/7*.0254*1000

In [3]:
print(y)

36.285714285714285


*The average intensity over 10 days is 36.285714285714285 millimerters per day.*

# __Question 3: Provide an example of string operations with the “+” operator. What is the “+” operator doing with strings? Explain in your own words.__


In [4]:
"CHICKEN"+"CHICKEN"+"CHICKEN"

'CHICKENCHICKENCHICKEN'

*In this example, the + operator is compressing together the words within the quotation marks*

# __Question 4: Provide an example of type conversion from numbers of type int and float into string and explain why it useful in combination with the + operator? The built-in Python function to use is ‘str()’.__

In [5]:
name="Emily"
times=13
tears=47.1
print(name + " just cried " + str(times) + " times and cried " + str(tears) + " tears while writing this assignment ")

Emily just cried 13 times and cried 47.1 tears while writing this assignment 


*This is useful in combonation with the + operator because this operator concatenates and can "string" together lines of code that are originally written in different forms with only one print function.*

# __Question 5: You get a table with 3-hour precipitation amounts: Sum up the numbers and store the result in a variable named total and print value of the variable.__

In [6]:
y=.1,1.5,4.7,3.5,8.5,.1,.2
total=sum(y)
print(total)

18.6


*The sum is 18.6mm*

# __Question 6: Use the function print()  to show the text “The rainfall total in 24 hours is:”, and the number (with ­­one print() function)__

In [7]:
print( " The rainfall total in 24 hours is " + str(total) + " mm ")

 The rainfall total in 24 hours is 18.6 mm 


# __Question 7:    Inform yourself about the function type() and try out the code and observe the results. What does the function type() tell you about the variable x and y?__

In [8]:
x=100
print(x)
print(type(x))

y=100/10
print(y)
print(type(y))

100
<class 'int'>
10.0
<class 'float'>


*The "type" function tells you the classification of x and y. The number represented by x is an integer and the number represented by y is a floating point number.*

# __Question 8: Try the following code:__
x=100

y=x

x=42

print (x,y)

# __What happened here? Explain the result by looking at how the sequence of variable assignment works in Python when we run the code.__


In [9]:
x=100
y=x
x=42
print(x,y)

42 100


*The result is in the format of an ordered pair. Since y is defined as x and x is defined as 42 afterward, the result is that 42 is printed first in the ordered pair.*

# __Question 9: Try the following code:__
 
x=100

y=x

y=42

print (x,y)
 
# __Compare this with the result from question 8. Do you understand the behavior of variable assignments?__

In [10]:
x=100
y=x
y=42
print(x,y)

100 42


*Since in this case the order is y=x followed by y=42, as opposed to x in the previous problem, the result is that 42 is printed second in the ordered pair.*

# __Question 10: We want to program a simple countdown from 10 to 0. First, assign the variable i the value 10.  Then subtract from i the value 1 (see code below). Repeat the code with the variable assignment i=i-1 so that you end with a value 0 in the last print statement. What is happening with variable i?__

In [11]:
i=10
print(i)
i=i-1
print(i)
i=i-1
print(i)
i=i-1
print(i)
i=i-1
print(i)
i=i-1
print(i)
i=i-1
print(i)
i=i-1
print(i)
i=i-1
print(i)
i=i-1
print(i)
i=i-1
print(i)

10
9
8
7
6
5
4
3
2
1
0


*The variable "i" is over and over again being deducted a single integer.*

# *__Conclusion__*

Variables are necessary in computer coding because we need symbols to take the place of ideas. They are also used to simplify problems, in order to repeat them again and again if need be. The variable is a tool to save information and use it again. It is important to be aware that Python distinguishes between types of data (such as int, float, and string) beacause it can easily become confusing if one does not know this. For example, the string "123" does not have the value of the integer one hundred and twenty three. It is simply three numbers strung together. It is also important to understand float, because in different operations such as addition and subtraction, if at least one of the numbers is float, the answer must be float. 